pkg/operator/nodekubeconfigcontroller/nodekubeconfigcontroller.go

```diff
@@ -152,6 +153,15 @@ func ensureNodeKubeconfigs(ctx context.Context, client coreclientv1.CoreV1Interf
 		requiredSecret.Annotations = map[string]string{}
 	}
 	requiredSecret.Annotations[annotations.OpenShiftComponent] = "kube-apiserver"
+	// Copy not-before/not-after annotations from systemAdminClientCert
+if systemAdminCredsSecret.Annotations != nil {
```

Comment:
in go reading from `nil` map works, no ?
xref: https://go.dev/play/p/hrmvBX3L2Q1

Reply:
`if len(systemAdminCredsSecret.Annotations[certrotation.CertificateNotBeforeAnnotation]) > 0` also works, right ?

Reply:
yes, the original code snippet is safe as-is regarding nil map reads.

Reply:
>len(systemAdminCredsSecret.Annotations[certrotation.CertificateNotBeforeAnnotation]) > 0

That would never reset an empty annotation if it was ever set, wouldn't it? Setting it to empty string is a common method to make controller set it again

Reply:
thanks